GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -281,8 +325,11 @@ - (void)libraryDataForKey:(nonnull NSString *)key
       // the implicit return value will be the block itself. The compiler doesn't detect this.
       if (newValue != nil && [newValue isKindOfClass:[NSData class]] && newValue.length) {
         NSError *newValueError;
-        [newValue writeToFile:dataPath options:NSDataWritingAtomic error:&newValueError];
-        if (newValueError) {
+        if ([newValue writeToFile:dataPath options:NSDataWritingAtomic error:&newValueError]) {
+          // Update storage size.
+          [self.sizeCalculator fileWithSize:data.length wasRemovedAtPath:dataPath];
```

Comment:
Instead of calling fileRemoved and added file, should we open up a new method named updateFileSizeAtDirectoryPath:WithDelta:.?

Reply:
I considered this approach but decided to use the two existing methods because of the following:
1. A method such `updateFileSizeAtDirectoryPath:WithDelta:` will require calculating the diff from the caller, so actually will be more difficult to use. Though the calculation will most likely be trivial it will have to be repeated on the caller side.
2. Benefits of the methods don't justify maintenance cost of additional API. 

Does it look reasonable to you?

Reply:
Ha.. That is true. Makes sense with this change.